---
title: "Bee microbiome gene expression"
author: "Sasha Mikheyev"
date: "11/7/2019"
output:
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = T)
```

```{r}
library(tidyverse)
library(sleuth)
#library(topGO) #annoyingly ovrrides select
library(readxl)
library(kableExtra)
```

```{r}
sample_id <- dir(file.path("data/kallisto"))
metadata <- read_xlsx("data/RNA_seq_sample_metafile.xlsx") 
t2g <- read_tsv("data/gene2isoform.txt.gz", col_names = c("gene_id", "target_id"))
genesAnnotation <- read_tsv("data/genes.txt", col_names = c("beebase", "target_id", "description"), col_types = "ccc") %>% group_by(target_id) %>% mutate(description = gsub(" isoform X\\d+","", description, perl = T), description = description[1]) %>% unique() 
go <- read_tsv("data/go.tsv", col_names = c("gene_id", "go"))
go <- split(go$go, go$gene_id) # convert to list for topGO

```

# Model incorporating all factors at once

```{r dge}
wald <- function(trt = "Tetracycline", contrast = "historystress_co_evolved") {
  dat <-  metadata %>% dplyr::filter(grepl(trt, treatment)) %>%  dplyr::select(sample, stress = time_stress, history) %>% mutate(path = paste0("./data/kallisto/", sample))
  so <- sleuth_prep(dat, extra_bootstrap_summary = T, target_mapping = t2g, aggregation_column = 'gene_id')
  plot_pca(so, color_by = 'stress') 
  so <- sleuth_fit(so, ~ stress + history, 'full')
  so <- sleuth_wt(so, contrast)
  results <- sleuth_results(so, test = contrast, pval_aggregate = F) %>% mutate(gene_id = stringr::str_trim(gene_id))
  return(left_join(results, genesAnnotation, by = c("gene_id" = "target_id"))  %>% dplyr::select(gene_id, target_id, qval, b, description))
}
```

Carry out the dge tests for each set of contrasts

```{r}
treatments <- c("Tetracycline", "Glyphosate", "Chlorothalonil")
# two contrasts: evolutionary history
# b is the effect size. 
#   b>0 is upregulated in coevolved vs. naive
#   b>0 is upregulated before stress vs. after stress
contrasts <- c('historystress_co_evolved', 'stressbefore_stress') 
dge <- list()
for (trt in treatments) 
  for (contrast in contrasts)
    dge[[trt]][[contrast]] <- wald(trt, contrast)
```

```{r printDGE, results='asis'}
for (trt in treatments) 
  for (contrast in contrasts) {
    print(dge[[trt]][[contrast]] %>% dplyr::filter(qval <= 0.05) %>%
      kable(caption = paste0(trt, ": ", ifelse(contrast == 'historystress_co_evolved', "effect of co-evolution", "effect of stress"))) %>% 
      kable_styling())
  }
```

```{r, echo = F}
# write dge results for later GO term analaysi
for (trt in treatments) 
  for (contrast in contrasts) 
    for (beta in c("up", "down"))
      {
      if (contrast == 'historystress_co_evolved') ctr = "coevolved"
      if (contrast == 'stressbefore_stress') ctr = "pre-stress"
      if (beta == "up") {
        write.table(file = paste0("./results/", trt, "_", ctr, "_upregulated.txt"), dge[[trt]][[contrast]] %>% dplyr::filter( qval <= 0.05 & b > 0))
      }
    else {
      write.table(file = paste0("./results/", trt, "_", contrast, "_downregulated.txt"), dge[[trt]][[contrast]] %>% dplyr::filter( qval <= 0.05 & b < 0))
    }
  }
```

```{r go, eval=FALSE, echo=FALSE}
dgeObject <- dge[["Tetracycline"]][['historystress_co_evolved']]

allGenes <- unique(t2g$gene_id)[unique(t2g$gene_id) %in% names(go)]
interestingGenes <- factor(as.integer(allGenes %in% dge[["Tetracycline"]][["stressbefore_stress"]]$gene_id))
names(interestingGenes) <- allGenes
GOdata <- new("topGOdata",
                description="Simple session",ontology="BP",
                allGenes=interestingGenes,
                annot=annFUN.gene2GO, gene2GO=go)
resultFisher <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
allRes <- GenTable(GOdata, classicFisher = resultFisher, orderBy = "classicKS", topNodes = 10)
allRes
```