---
title: "Bee microbiome gene expression"
author: "Sasha Mikheyev"
date: "11/7/2019"
output:
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = T)
```

```{r}
library(tidyverse)
library(sleuth)
#library(topGO) #annoyingly ovrrides select
library(readxl)
library(kableExtra)
```

```{r}
sample_id <- dir(file.path("data/kallisto"))
metadata <- read_xlsx("data/RNA_seq_sample_metafile.xlsx") 
t2g <- read_tsv("data/gene2isoform.txt.gz", col_names = c("gene_id", "target_id"))
genesAnnotation <- read_tsv("data/genes.txt", col_names = c("beebase", "target_id", "description"), col_types = "ccc") %>% group_by(target_id) %>% mutate(description = gsub(" isoform X\\d+","", description, perl = T), description = description[1]) %>% unique() 
#go <- read_tsv("data/go.tsv", col_names = c("gene_id", "go"))
#go <- split(go$go, go$gene_id) # convert to list for topGO

stressGenes <- rbind(read_xlsx("data/doublet.xlsx") %>% mutate(gene_id = as.character(NCBI_ID)) %>% dplyr::select(gene_id, pvalue), 
  read_xlsx("data/gene stress list for RNA studies.xlsx", range = "B3:B39", col_names = "gene_id", col_types = "text") %>% mutate(pvalue = NA)) %>% group_by(gene_id)

```

# Model incorporating all factors at once

```{r dge}
wald <- function(trt = "Tetracycline", contrast = "historystress_co_evolved") {
  dat <-  metadata %>% dplyr::filter(grepl(trt, treatment)) %>%  dplyr::select(sample, stress = time_stress, history) %>% mutate(path = paste0("./data/kallisto/", sample))
  so <- sleuth_prep(dat, extra_bootstrap_summary = T, target_mapping = t2g, aggregation_column = 'gene_id')
  plot_pca(so, color_by = 'stress') 
  so <- sleuth_fit(so, ~ stress + history, 'full')
  so <- sleuth_wt(so, contrast)
  results <- sleuth_results(so, test = contrast, pval_aggregate = F) %>% mutate(gene_id = stringr::str_trim(gene_id))
  return(left_join(results, genesAnnotation, by = c("gene_id" = "target_id"))  %>% dplyr::select(gene_id, target_id, pval, qval, b, description))
}
```

Carry out the dge tests for each set of contrasts

```{r}
treatments <- c("Tetracycline", "Glyphosate", "Chlorothalonil")
# two contrasts: evolutionary history
# b is the effect size. 
#   b>0 is upregulated in coevolved vs. naive
#   b>0 is upregulated before stress vs. after stress
contrasts <- c('historystress_co_evolved', 'stressbefore_stress') 
dge <- list()
for (trt in treatments) 
  for (contrast in contrasts)
    dge[[trt]][[contrast]] <- wald(trt, contrast)
```

```{r printDGE, results='asis'}
for (trt in treatments) 
  for (contrast in contrasts) {
    print(dge[[trt]][[contrast]] %>% dplyr::filter(qval <= 0.05) %>%
      kable(caption = paste0(trt, ": ", ifelse(contrast == 'historystress_co_evolved', "effect of co-evolution", "effect of stress"))) %>% 
      kable_styling())
  }
```

```{r saveDGE, echo=F}
for (trt in treatments) 
  for (contrast in contrasts) {
    title <- paste0("data/dge/",trt, " ", ifelse(contrast == 'historystress_co_evolved', "effect of co-evolution", "effect of stress"), ".txt")
    write.table(file = title, dge[[trt]][[contrast]], row.names = F)
  }
```


```{r, echo = F}
# write dge results for later GO term analaysi
for (trt in treatments) 
  for (contrast in contrasts) 
    for (beta in c("up", "down"))
      {
      if (contrast == 'historystress_co_evolved') ctr = "coevolved"
      if (contrast == 'stressbefore_stress') ctr = "pre-stress"
      if (beta == "up") {
        write.table(file = paste0("./results/", trt, "_", ctr, "_upregulated.txt"), dge[[trt]][[contrast]] %>% dplyr::filter( qval <= 0.05 & b > 0))
      }
    else {
      write.table(file = paste0("./results/", trt, "_", contrast, "_downregulated.txt"), dge[[trt]][[contrast]] %>% dplyr::filter( qval <= 0.05 & b < 0))
    }
  }
```

## Comparing co-evolved vs control bees before stress

```{r dge_treatment}
# one contrast: evolutionary history
# b is the effect size. 
#   b>0 is upregulated in coevolved vs. naive

waldCoevolved <- function(trt = "Tetracycline") {
  dat <-  metadata %>% dplyr::filter((treatment == trt | history == "control") & time_stress == "before_stress") %>%  dplyr::select(sample, history) %>% mutate(path = paste0("./data/kallisto/", sample))
  so <- sleuth_prep(dat, extra_bootstrap_summary = T, target_mapping = t2g, aggregation_column = 'gene_id')
  so <- sleuth_fit(so, ~ history, 'full')
  so <- sleuth_wt(so, "historystress_co_evolved")
  results <- sleuth_results(so, test = "historystress_co_evolved", pval_aggregate = F) %>% mutate(gene_id = stringr::str_trim(gene_id))
  return(left_join(results, genesAnnotation, by = c("gene_id" = "target_id"))  %>% dplyr::select(gene_id, target_id, pval, qval, b, description))
}
```


```{r dgeCoevolved, results='asis'}
dgeCoevolution <- list()
for (trt in treatments) {
  dgeCoevolution[[trt]] <- waldCoevolved(trt)
  print(dgeCoevolution[[trt]] %>% dplyr::filter(qval <= 0.05) %>% kable(caption = paste("Bee genes upregulated as a result of ", trt, "coevolution")) %>% 
      kable_styling())
  title <- paste0("data/dge/",trt, " exposure.txt")
  write.table(file = title, dgeCoevolution[[trt]], row.names = F)
}
```


## What happens to immune genes in the course of adaptation?
```{r doublet}
(immuneGenes <- rbind(
  stressGenes %>% left_join(dge[["Tetracycline"]][["historystress_co_evolved"]], by =  "gene_id" ) %>% mutate(treatment = "Tetracycline"),
  stressGenes %>% left_join(dge[["Glyphosate"]][["historystress_co_evolved"]], by =  "gene_id" ) %>% mutate(treatment = "Glyphosate"),
  stressGenes %>% left_join(dge[["Chlorothalonil"]][["historystress_co_evolved"]], by =  "gene_id" ) %>% mutate(treatment = "Chlorothalonil")
) )%>% ggplot(aes(b, color=treatment)) + geom_density() + theme_minimal() + geom_vline(xintercept = 0)

immuneGenes %>% ggplot(aes(treatment, b))  + geom_violin() + theme_minimal() + geom_hline(yintercept= 0, color = "red") + stat_summary(fun.y = mean, geom = "point") 

TetVsChlor <- dge[["Tetracycline"]][["historystress_co_evolved"]] %>% dplyr::select(gene_id, target_id, b, qval) %>% inner_join(stressGenes, by =  "gene_id" ) %>%  inner_join(dge[["Chlorothalonil"]][["historystress_co_evolved"]], by =  "target_id" ) 
with(TetVsChlor, cor.test(b.x, b.y, method = "s"))
TetVsChlor %>% ggplot(aes(b.x, b.y)) + geom_hex()

```

```{r go, eval=FALSE, echo=FALSE}
dgeObject <- dge[["Tetracycline"]][['historystress_co_evolved']]

allGenes <- unique(t2g$gene_id)[unique(t2g$gene_id) %in% names(go)]
interestingGenes <- factor(as.integer(allGenes %in% dge[["Tetracycline"]][["stressbefore_stress"]]$gene_id))
names(interestingGenes) <- allGenes
GOdata <- new("topGOdata",
                description="Simple session",ontology="BP",
                allGenes=interestingGenes,
                annot=annFUN.gene2GO, gene2GO=go)
resultFisher <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
allRes <- GenTable(GOdata, classicFisher = resultFisher, orderBy = "classicKS", topNodes = 10)
allRes
```


### Verifying with pre-treatment samples only

We're going to remove post-stress samples to see if we have a similar pattern to the whole model.

```{r}
wald2 <- function(trt = "Tetracycline") {
  dat <-  metadata %>% dplyr::filter(grepl(trt, treatment) & time_stress == "before_stress") %>%  dplyr::select(sample, history) %>% mutate(path = paste0("./data/kallisto/", sample))
  so <- sleuth_prep(dat, extra_bootstrap_summary = T, target_mapping = t2g, aggregation_column = 'gene_id')
  so <- sleuth_fit(so, ~ history , 'full')
  so <- sleuth_wt(so, 'historystress_co_evolved')
  results <- sleuth_results(so, test = "historystress_co_evolved", pval_aggregate = F) %>% mutate(gene_id = stringr::str_trim(gene_id))
  return(left_join(results, genesAnnotation, by = c("gene_id" = "target_id"))  %>% dplyr::select(gene_id, target_id, qval, b, description))
}
```

```{r}
dge2 <- list()
for (trt in c("Tetracycline", "Glyphosate")) 
    dge2[[trt]] <- wald2(trt)

(immuneGenes2 <- rbind(
  stressGenes %>% left_join(dge2[["Tetracycline"]], by =  "gene_id" ) %>% mutate(treatment = "Tetracycline"),
  stressGenes %>% left_join(dge2[["Glyphosate"]], by =  "gene_id" ) %>% mutate(treatment = "Glyphosate")
) )%>% ggplot(aes(b, color=treatment)) + geom_density() + theme_minimal() + geom_vline(xintercept = 0)

immuneGenes2 %>% ggplot(aes(treatment, b))  + geom_violin() + theme_minimal() + geom_hline(yintercept= 0, color = "red") + stat_summary(fun.y = mean, geom = "point") 
```

Seems that the overall test partitions the variance pretty well and we get the same pattern with pre-stress treatmentsm, too.