---
title: "R_microbiome_analysis_final"
author: "Vienna"
date: "4/25/2020"
output: 
  pdf_document: default
  keep_md: true
  html_document: default
self_contained: no
thumbnails: no
keep_md: yes
---

## Summary
The experiment aimed to study the response of honey bee microbiome to three chemicals across host generations and to detect potential microbiome mediated effects on host phenotypes. 
## Methods
We reared adult bees under controlled lab conditions and inoculated them with a natural bee microbiome (start_pool). Bees were orally stressed with either Tetracycline, Glyphosate, Chloropthalonil or no stressor (control). Three cages per treatment were used. Both, the stress-exposed and control microbiomes were transferred to the next cycle (cage to cage transfer) which was handled as before. The third cycle aims to study effects of the pre-exposed microbiome on phenotypes of naive bee hosts in comparison to control microbiomes. For that we transferred the microbiomes and let them be established in the bee hosts without any stress factor contact. We finally applied high amounts of chemicals to bees with a control microbiome or a pre-exposed microbiome. Bee samples for 16S sequencing has been snap-frozen after cycle 1, cycle 2, cycle 3 BEFORE high stress and cycle 3 AFTER high stress. In addition, the macerated gut pools for transferring microbiome has been saved (start_microbiome as well as pool from each cage to cage transfer). We sequenced the V3-V4 region of the 16S region. Two DNA mock samples from ZymoResearch have been sequenced and named as "positive_control".

### Data and metadata

Overview over all experimental variables:

**Experimental variables find in metadata file**:  

*sample_type* <- microbiome_transfer or single bee
*treatment* <- treatment used (control or which toxin) and additional information if microbiome transfer (e.g. Control_transfer), single bee (e.g. Control)
*treatment2* <- only treatment, not indicating sample type
*cage* <- cage number (three cages per treatment have been used)
*treatment_cage* <- combined information of treatment and cage number (e.g. Control_2)
*date* <- date of sampling during experiment
*cycle* <- experimental cycle (cycle 1, cycle 2, cycle 3 before stress, cycle 3 after stress)
*treatment_cycle* <- experimental cycle in combination with treatment information
*treatment_cycle2*<- experimental cycle in combination with treatment  plus sample type information

```{r setup, include = FALSE}
library(RSQLite)
library(DECIPHER)
library(DESeq2)
library(gridExtra)
library(ggnewscale) ##new_scale()
library(ggpubr) # ggarrange()
library(ggthemes)
library(grid) # rasterGrob()
library(vegan)
library(phyloseq)
library(png)
library(qgraph) # graph layout
library(ggsignif)
library(btools)
library(RColorBrewer)
library(tidyverse) # includes: ggplot2, dplyr, tidyr, readr, purr, tibble, stringr, forcats
library(knitr)

knitr::opts_chunk$set(warning=FALSE, message=FALSE)

set.seed(42)
rm(list = ls())

# Plotting options
plot_res = 300 # DPI of PNG graphs
col_cembio <- c(rgb(181, 6, 155, 255, max = 255), rgb(64, 118, 253, 255, max = 255))
col_c12 <- brewer.pal(12, "Set3")
theme_set(theme_bw())

# Analysis switches
do_tech_analysis <- TRUE
```


### Statistics on survival data of bees with pre-exposed microbiomes vs respective controls under high chemical stress for main experiment

```{r survival_data}
tetra <- read.table("R_microbiome_data_files/tetra cycle 3 day 5 to 6 survival.txt", header = TRUE)
fisher.test(tetra, alternative = "two.sided")

chloro <- read.table("R_microbiome_data_files/chloro cycle 3 day 5 to 6 survival.txt", header = TRUE)
fisher.test(chloro, alternative = "two.sided")

glypho <- read.table("R_microbiome_data_files/glypho cycle 3 day 5 to 7 survival.txt", header = TRUE)
fisher.test(glypho, alternative = "two.sided")
```
Glyphosate-exposed microbiomes did not significantly affect the survival of bees under high glyphosate stress, chlorothalonil-exposed microbiomes mediated protection and tetracycline-exposed microbiomes lead to higher mortality


### Additional chlorothalonil experiments to figure out protective mechanisms of chlorothalonil-exposed microbiomes on bee survival. Statistics on survival data of bees with added filtered pre-exposed gut extract and added chlorothalonil vs respective controls under high chemical stress.
```{r survival_data_chloro_controls}
survive <- read.table("R_microbiome_data_files/Fisher_test_filtered_Chloro_control_exp.txt", header = TRUE)
fisher.test(survive, alternative = "two.sided")


survive2 <- read.table("R_microbiome_data_files/Fisher_test_added_Chloro_control_exp.txt", header = TRUE)
fisher.test(survive2, alternative = "two.sided")
``` 
While filtered pre-exposed gut solution did improve later survival under high chlorothalonil stress, direct addition of chlorothalonil did not.


## 16S data


```{r load}
#read in otu table
otu_table=read.csv("R_microbiome_data_files/Svtab1.csv",sep=",",row.names=1)
otu_table=as.matrix(otu_table)

#read in taxonomy
taxonomy=read.csv("R_microbiome_data_files/taxonomy_modify.csv",sep=",",row.names=1)
taxonomy <- cbind(taxonomy, ASV = paste0("ASV", sprintf("%04d", 1:nrow(taxonomy))))
taxonomy=as.matrix(taxonomy)

metatable <- read.delim("R_microbiome_data_files/metadata_reorder3.txt") 
#View(metatable)
row.names(metatable) <- metatable[[1]]
metatable<- metatable[,(-1)]
META <- sample_data(metatable)

phy_tree <- read_tree("R_microbiome_data_files/rooted_tree.nwk")

#import as phyloseq objects
OTU=otu_table(otu_table,taxa_are_rows=TRUE)
TAX=tax_table(taxonomy)

#create phyloseq object
ps1<- phyloseq(OTU, TAX, META, phy_tree)

# change taxonomy header 
colnames(tax_table(ps1)) <- c(D0 = "Kingdom", D1 = "Phylum", D2 = "Class", 
                              D3 = "Order", D4 = "Family", D5 = "Genus", D6 = "Species", ASV = "ASV")

#The total number of ASVs in the whole dataset is 
length(taxa_names(ps1))

#get rid of things we do not want
ps1 = subset_taxa(ps1, Kingdom =="Bacteria")
ps1 <- prune_taxa(taxa_sums(ps1) > 0, ps1)
ps1<-subset_taxa(ps1, (Order!="Chloroplast"))
ps1<-subset_taxa(ps1, (Family!="Mitochondria"))
length(taxa_names(ps1))
#reduces total taxa numbers from 1717 to 1167 (minus 550)

# mean, max and min of sample read counts
smin <- min(sample_sums(ps1))
smean <- mean(sample_sums(ps1))
smax <- max(sample_sums(ps1))
# printing the results
cat("The minimum sample read count is:",smin)
cat("The average sample read count is:",smean)
cat("The maximum sample read count is:",smax)
```


### Plot rarefaction curves
```{r rarefaction_curve, fig.heigt=5, fig.width=12}
rare<- ps1
rare2 = subset_samples(rare, treatment2 != "positive_control")
smin <- min(sample_sums(rare2))
cat("The minimum sample read count is:",smin)

# rarefy
set.seed(42)
ps.rarefied = rarefy_even_depth(rare2, rngseed=1, sample.size=12351, replace=F)
library(ranacapa)
cbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7","maroon","khaki4")

ggrare(ps.rarefied, step = 100, se = FALSE, color="treatment2")+ scale_colour_manual(values=cbPalette)+geom_line(size=1)+ theme_bw()+theme(axis.line = element_line(colour = "black"))+ theme(legend.title = element_blank(),
                legend.background = element_blank(),
                legend.key = element_blank()) + theme(text = element_text(size = 22))+theme(strip.text = element_text(face="bold", size=20))+ theme(axis.title.x=element_blank())+ggtitle("Rarefaction even depth")+theme(plot.title = element_text(hjust = 0.5, size=20, face="italic"))+theme(legend.text=element_text(size=17, face="italic"))
ggsave("R_microbiome_figures/supp_rarefaction_sub.png", height = 5, width = 12)
```

*** Supplementary figure: Rarefaction curves were used as a qualitative method to estimate the species richness as a function of sequencing depth for all samples. Rarefaction curves reached their asymptotes for all samples, suggesting that saturation in sequencing was achieved.

### Positive controls / mock community

The experiment contained two positive control samples with DNA from the ZymoBIOMICSâ„¢ Microbial
Community Standard (Zymo Research).


#### Plot taxa in Mock control samples

```{r mock, eval = do_tech_analysis}
ps_mock <- ps1
ps_mock <- subset_samples(ps_mock, treatment == "positive_control")
ps_mock <- prune_taxa(taxa_sums(ps_mock) > 0, ps_mock)
#transform to proportional data
ps_mock2 <- transform_sample_counts(ps_mock, function(OTU) {OTU / sum(OTU)})
ps_mock2 <- tax_glom(ps_mock2, taxrank = 'Genus')
low1pc_reads <- max(taxa_sums(ps_mock2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_mock2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_mock2)[low1pc_indices[1]] <- "other"
ps_merge <- merge_taxa(ps_mock2, low1pc_indices, "other")
tax_table(ps_merge)["other", ] <- c("other")

getPalette = colorRampPalette(brewer.pal(10, "Set3"))
speciesList = unique(tax_table(ps_merge)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList

ps_df <- psmelt(ps_merge)

ps_df_sum <- ps_df %>%
  group_by(Sample,Genus) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(Sample) %>%
  mutate(Relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE)) %>%
  droplevels()
levels(ps_df_sum$Genus)

mock_plot<-ggplot(data = ps_df_sum, aes(x = Sample, y = Relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_y_continuous(expand = c(0,0))+ scale_fill_manual(values= speciesPalette)+ylab("relative abundance")
#mock_plot<-mock_plot+labs(title="Mock samples")
mock_plot<-mock_plot+ theme(legend.text=element_text(size=9,face="italic"))
mock_plot<-mock_plot+theme(axis.text.x = element_text(size=12, face="bold"))+ theme(axis.title.x=element_blank())+ theme(legend.title=element_blank())+theme(axis.text.y = element_text(size=12, face="bold"))+ theme(axis.title.y=element_text(size=13, face="bold"))
mock_plot2<-mock_plot+theme(plot.title = element_text(face = "bold", color="grey40",size = (17)))+theme(plot.title = element_text(hjust = 0.5))
ps_df_sum
```

```{r figure_S1, fig.align = "center",eval = do_tech_analysis}
mock_comp <- "R_microbiome_data_files/mock_composition_ZymoResearch.png"
figure_S1 <- "R_microbiome_figures/Supp_mock_composition_comparison.png"
png(figure_S1, 7 * plot_res, 4 * plot_res, res = plot_res)
ggarrange(mock_plot2, rasterGrob(readPNG(mock_comp)),labels = c("A", "B"))
invisible(dev.off())
knitr::include_graphics(figure_S1, dpi = plot_res)
```
All 8 Mock bacterial taxa correctly detected. 
**Figure S1: Comparison of the mock community with the theoretical proportions.**
The two mock samples sequenced in this study **(A)** show no big difference to the expected
theoretical proportions of the reference **(B, bar with label *theoretical*)**. The three other bacterial taxa belong to honey bee core microbiome (Bartonella, Gilliamella and Snodgrassella). The relative abundance of these three taxa across the two Mock samples account for 0.23% of all reads, representing neglectable cross-contamination during sequencing.


#### comparing difference between methods
We sequenced to types of samples: whole bee abdomen and the mix of three macerated guts for cage to cage transfers after each cycle
Therefore, we need to test if these samples are different due to the differences in the methods prior extracting before deciding if we include all or not.
Use PERMANOVA on bray-curtis dissimilarities using proportional transformed abundance data
```{r}
set.seed(42)
methods = ps1
methods <- transform_sample_counts(methods, function(OTU) {OTU / sum(OTU)})

Control = subset_samples(methods, treatment2 == "Control")
Control <- prune_taxa(taxa_sums(Control) > 0, Control)
C_metadata <- as(sample_data(Control), "data.frame")
adonis(distance(Control, method="bray") ~ sample_type,data = C_metadata,perm=999)

Tetra = subset_samples(methods, treatment2 == "Tetracycline")
Tetra <- prune_taxa(taxa_sums(Tetra) > 0, Tetra)
T_metadata <- as(sample_data(Tetra), "data.frame")
adonis(distance(Tetra, method="bray") ~ sample_type,data = T_metadata, perm=999)

Glypho = subset_samples(methods, treatment2 == "Glyphosate")
Glypho <- prune_taxa(taxa_sums(Glypho) > 0, Glypho)
G_metadata <- as(sample_data(Glypho), "data.frame")
adonis(distance(Glypho, method="bray") ~ sample_type,data = G_metadata, perm=999)

Chloro = subset_samples(methods, treatment2 == "Chlorothalonil")
Chloro <- prune_taxa(taxa_sums(Chloro) > 0, Chloro)
Ch_metadata <- as(sample_data(Chloro), "data.frame")
adonis(distance(Chloro, method="bray") ~ sample_type,data = Ch_metadata, perm=999)
```
No significant difference between gut pools or whole bees in any treatment.


##  Alpha diversity 

```{r alpha_Shannon, fig.heigt=13, fig.width=8}
alpha<- ps1
#pick cycle 1, 2 and 3 before stress
alpha1 = subset_samples(alpha, cycle=="cycle_one" | cycle=="cycle_two" | cycle == "cycle_three_before_stress")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha1,sample.size=12351, replace=FALSE, rngseed = 1) 
alpha1_ra<-prune_taxa(taxa_sums(alpha1_ra) > 0, alpha1_ra)

sample_data(alpha1_ra)$cycle<-factor(sample_data(alpha1_ra)$cycle,levels=c("cycle_one","cycle_two","cycle_three_before_stress"), labels=c("Cycle one","Cycle two","Cycle three"))
levels(sample_data(alpha1_ra)$cycle)

sample_data(alpha1_ra)$treatment3<-factor(sample_data(alpha1_ra)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"))
levels(sample_data(alpha1_ra)$treatment3)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

alpha_meas = c("Shannon")
p2 <- plot_richness(alpha1_ra,"treatment3",color="treatment3", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment3, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Control", "Chlorothalonil"), c("Control", "Glyphosate"),c("Control", "Tetracycline")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(3.7,3.7,3.9), margin_top = 0.1, textsize=6, test = wilcox.test)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(1.7,4)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Shannon alpha diversity")
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 17))+theme(strip.text = element_text(face="bold", size=15, color="gray40"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_blank())
Shannon2<-p2+ facet_grid( ~cycle, scales="free_x",space="free")+theme(axis.ticks.x=element_blank())+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
Shannon2
ggsave("R_microbiome_figures/alpha_Shannon.png", height = 3.5, width = 6.5)
```
***The Shannon index  captures  information  about  both  the  species  richness  (number  of species) and the relative abundances of the species. Increased Shannon index means increased species diversity.

### make statistical comparisons on Shannon alpha diversity index - comparing treatments against respective controls in each cycle

```{r warning=FALSE}
alpha<- ps1
alpha1 = subset_samples(alpha, cycle=="cycle_one" | cycle=="cycle_two" | cycle == "cycle_three_before_stress")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha1,sample.size=12351, replace=FALSE, rngseed = 1) 

results = estimate_richness(alpha1_ra, measures = 'Shannon')
d = sample_data(alpha1_ra)

# calculate wilcox-test
Control_1 = results[d[,'treatment_cycle3'] == 'Control_cycle_1',]
Tetracycline_1 = results[d[,'treatment_cycle3'] == 'Tetracycline_cycle_1',]
Glyphosate_1 = results[d[,'treatment_cycle3'] == 'Glyphosate_cycle_1',]
Chlorothalonil_1 = results[d[,'treatment_cycle3'] == 'Chlorothalonil_cycle_1',]
wilcox.test(Control_1, Chlorothalonil_1)
wilcox.test(Control_1, Glyphosate_1)
wilcox.test(Control_1, Tetracycline_1)

pvalues<-c(0.0596,0.002914,0.004396)
p.adjust(pvalues,method="fdr")

#cycle 2
Control_2 = results[d[,'treatment_cycle3'] == 'Control_cycle_2',]
Tetracycline_2 = results[d[,'treatment_cycle3'] == 'Tetracycline_cycle_2',]
Glyphosate_2 = results[d[,'treatment_cycle3'] == 'Glyphosate_cycle_2',]
Chlorothalonil_2 = results[d[,'treatment_cycle3'] == 'Chlorothalonil_cycle_2',]
wilcox.test(Control_2, Chlorothalonil_2)
wilcox.test(Control_2, Glyphosate_2)
wilcox.test(Control_2, Tetracycline_2)

pvalues<-c(0.671,0.9279,7.396e-07)
p.adjust(pvalues,method="fdr")

#cycle 3
control_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'control_cycle_3_before_stress',]
Tetracycline_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'Tetracycline_cycle_3_before_stress',]
Glyphosate_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'Glyphosate_cycle_3_before_stress',]
Chlorothalonil_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'Chlorothalonil_cycle_3_before_stress',]
wilcox.test(control_cycle_3_before_stress, Chlorothalonil_cycle_3_before_stress)
wilcox.test(control_cycle_3_before_stress, Glyphosate_cycle_3_before_stress)
wilcox.test(control_cycle_3_before_stress, Tetracycline_cycle_3_before_stress)

pvalues<-c(0.641,0.000385,2.124e-08)
p.adjust(pvalues,method="fdr")
```
Under tetracycline the Shannon alpha diversity was significantly affected in all three cycles. Glyphosate significantly affected the alpha diversity in cycle 1 and 3 and chlorothalonil did not had any significant effect.

### observed species numbers - alpha diversity
```{r alpha_Observed, fig.heigt=13, fig.width=8}
alpha<- ps1
alpha1 = subset_samples(alpha, treatment != "hive_bee" & treatment != "positive_control"& treatment != "start_microbiome"& cycle != "cycle_three_after_stress")
#rarefy to even numbers
smin <- min(sample_sums(alpha1))
cat("The minimum sample read count is:",smin)
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha1,sample.size=12351, replace=FALSE, rngseed = 1) 

sample_data(alpha1_ra)$cycle<-factor(sample_data(alpha1_ra)$cycle,levels=c("cycle_one","cycle_two","cycle_three_before_stress"), labels=c("Cycle one","Cycle two","Cycle three"))
levels(sample_data(alpha1_ra)$cycle)

sample_data(alpha1_ra)$treatment3<-factor(sample_data(alpha1_ra)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"))
levels(sample_data(alpha1_ra)$treatment3)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

alpha_meas = c("Observed")
p2 <- plot_richness(alpha1_ra,"treatment3",color="treatment3", measures=alpha_meas)
p2$layers <- p2$layers[-1]
p2<-p2 + geom_boxplot(data=p2$data, aes(x=treatment3, y=value),show_guide=FALSE, alpha=0.1)+geom_point(position = position_jitter(w = 0.2, h = 0.1),show_guide=FALSE) +
  geom_signif(comparisons=list(c("Control", "Chlorothalonil"), c("Control", "Glyphosate"),c("Control", "Tetracycline")),tip_length = 0.02,vjust=0.5, color=("grey20"),map_signif_level=sigFunc, y_position = c(70,77,81), margin_top = 0.1, textsize=6, test = wilcox.test)

p2 <- p2 + guides(fill=guide_legend(title=element_blank()))+ylim(17,84)
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank()) +ylab("Observed alpha diversity")
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 17))+theme(strip.text = element_text(face="bold", size=15, color="gray40"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_blank())
p2<-p2+ facet_grid( ~cycle,scales="free_x",space="free")+theme(axis.ticks.x=element_blank())+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
Observed<-p2+ geom_point(size = -1, shape = 15,fill="treatment3") +geom_boxplot(show.legend = FALSE)+guides(color = guide_legend(override.aes = list(size = 7,fill="treatment3",alpha=0.5)))+ theme(axis.title.y=element_text(size=14))
Observed
ggsave("R_microbiome_figures/Supp_alpha_Observed.png", height = 3.5, width = 6.5)
```
*** Supplementary figure Observed alpha diversity
The observed plot shows the difference in detected OTUs. 

### make statistical comparisons on Observed species numbers - comparing treatments against respective controls in each cycle
```{r warning=FALSE}
alpha<- ps1
alpha1 = subset_samples(alpha, cycle=="cycle_one" | cycle=="cycle_two" | cycle == "cycle_three_before_stress")
#rarefy to even numbers
set.seed(1)  
alpha1_ra <- rarefy_even_depth(alpha1,sample.size=12351, replace=FALSE, rngseed = 1) 

results = estimate_richness(alpha1_ra, measures = 'Observed')
d = sample_data(alpha1_ra)

# calculate wilcox-test
Control_1 = results[d[,'treatment_cycle3'] == 'Control_cycle_1',]
Tetracycline_1 = results[d[,'treatment_cycle3'] == 'Tetracycline_cycle_1',]
Glyphosate_1 = results[d[,'treatment_cycle3'] == 'Glyphosate_cycle_1',]
Chlorothalonil_1 = results[d[,'treatment_cycle3'] == 'Chlorothalonil_cycle_1',]
wilcox.test(Control_1, Chlorothalonil_1)
wilcox.test(Control_1, Glyphosate_1)
wilcox.test(Control_1, Tetracycline_1)

pvalues<-c(0.139,0.0184,0.0111)
p.adjust(pvalues,method="fdr")

#cycle 2
Control_2 = results[d[,'treatment_cycle3'] == 'Control_cycle_2',]
Tetracycline_2 = results[d[,'treatment_cycle3'] == 'Tetracycline_cycle_2',]
Glyphosate_2 = results[d[,'treatment_cycle3'] == 'Glyphosate_cycle_2',]
Chlorothalonil_2 = results[d[,'treatment_cycle3'] == 'Chlorothalonil_cycle_2',]
wilcox.test(Control_2, Chlorothalonil_2)
wilcox.test(Control_2, Glyphosate_2)
wilcox.test(Control_2, Tetracycline_2)

pvalues<-c(0.222,0.477,3.449e-05)
p.adjust(pvalues,method="fdr")

#cycle 3
control_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'control_cycle_3_before_stress',]
Tetracycline_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'Tetracycline_cycle_3_before_stress',]
Glyphosate_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'Glyphosate_cycle_3_before_stress',]
Chlorothalonil_cycle_3_before_stress = results[d[,'treatment_cycle3'] == 'Chlorothalonil_cycle_3_before_stress',]
wilcox.test(control_cycle_3_before_stress, Chlorothalonil_cycle_3_before_stress)
wilcox.test(control_cycle_3_before_stress, Glyphosate_cycle_3_before_stress)
wilcox.test(control_cycle_3_before_stress, Tetracycline_cycle_3_before_stress)

pvalues<-c(0.072,0.038,9.364e-06)
p.adjust(pvalues,method="fdr")
```
Numbers of observed species is significantly different under tetracycline to the control in all three cycles, while glyphosate affected the observed species number significantly in cycle 1 and chlorothalonil had no significant effect at any time point.


# Beta diversity

### ordination plots
```{r NMDS_man, fig.heigt=7.5, fig.width=10}
ord2<- ps1
ord2 = subset_samples(ord2, cycle=="cycle_one" | cycle=="cycle_two" | cycle == "cycle_three_before_stress")

ord2 <- transform_sample_counts(ord2, function(OTU) {OTU / sum(OTU)})
sample_data(ord2)$cycle<-factor(sample_data(ord2)$cycle,levels=c("cycle_one","cycle_two", "cycle_three_before_stress"), labels=c("Cycle one","Cycle two","Cycle three"))
levels(sample_data(ord2)$cycle)

sample_data(ord2)$treatment3<-factor(sample_data(ord2)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"))
levels(sample_data(ord2)$treatment3)

ord2 <- prune_taxa(taxa_sums(ord2) > 0, ord2)
set.seed(42)
dist<-phyloseq::distance(ord2, method="bray") 
ordination = ordinate(ord2, method="NMDS", distance=dist)
cat("stress is:", ordination$stress)
p2 <- plot_ordination(ord2, ordination, color="treatment3") +geom_point(size=2.5)
p2 <- p2 + guides(fill=guide_legend(title=element_blank()))
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank())
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 16.5))+theme(strip.text = element_blank())
#+theme(strip.text = element_text(face="bold", size=14, color="gray40"))+ theme(axis.title.x=element_blank())+theme(axis.text.x = element_blank())
p2<-p2+ facet_wrap( ~cycle,scales="free_x")+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
NMDS_main<-p2+stat_ellipse(aes(group = treatment3),size=1.1, alpha=0.5)+ theme(panel.spacing.x=unit(5.5, "lines"))+coord_cartesian(xlim = c(-0.9, 0.3))+ theme(axis.title.y = element_text(margin = margin(t = 0, r = 0, b = 0, l = 20)))
NMDS_main
ggsave("R_microbiome_figures/NMDS_man.png", height = 5, width = 10)
```

```{r PCOA_man_supp, fig.heigt=7.5, fig.width=10}
ord2<- ps1
ord2 = subset_samples(ord2, cycle=="cycle_one" | cycle=="cycle_two" | cycle == "cycle_three_before_stress")
ord2 <- transform_sample_counts(ord2, function(OTU) {OTU / sum(OTU)})
sample_data(ord2)$cycle<-factor(sample_data(ord2)$cycle,levels=c("cycle_one","cycle_two", "cycle_three_before_stress"), labels=c("Cycle one","Cycle two","Cycle three"))
levels(sample_data(ord2)$cycle)

sample_data(ord2)$treatment3<-factor(sample_data(ord2)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"))
levels(sample_data(ord2)$treatment3)

set.seed(42)
dist<-phyloseq::distance(ord2, method="bray") 
ordination = ordinate(ord2, method="PCoA", distance=dist)
p2 <- plot_ordination(ord2, ordination, color="treatment3") +geom_point(size=3.5)+scale_colour_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

p2 <- p2 + guides(fill=guide_legend(title=element_blank()))
p2 <- p2 + theme(plot.title = element_text(hjust = 0.5))+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p2 <- p2+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ theme(panel.background = element_blank())
p2 <- p2+theme(axis.line = element_line(colour = "black"))
p2<- p2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank())
p2 <- p2+ theme(panel.border = element_blank())
p2<-p2 + theme(text = element_text(size = 17))+theme(strip.text = element_text(face="bold", size=14, color="gray40"))+ theme(axis.title.x=element_blank())
p2<-p2+ facet_grid( ~cycle,scales="free_x",space="free")+theme(legend.position="bottom")+theme(strip.background =element_rect(fill="white", color="white"))
p2<-p2+stat_ellipse(aes(group = treatment3),size=1.1)+ theme(panel.spacing.x=unit(5.5, "lines"))
p2+scale_y_continuous(breaks = seq(0.0, 0.5))
ggsave("R_microbiome_figures/PCoA_man.png", height = 5, width = 9)
```

## Beta diversity stats

#### test difference between treatments and respective controls in the three cycles to statistically verify microbial community compositional difference seen in ordination plots 
```{r}
set.seed(42)
compare=ps1
compare1 <- transform_sample_counts(compare, function(OTU) {OTU / sum(OTU)})

cycle1 = subset_samples(compare1, cycle == "cycle_one")
cycle1 <- prune_taxa(taxa_sums(cycle1) > 0, cycle1)
cycle1.1 <- as(sample_data(cycle1), "data.frame")

cycle2 = subset_samples(compare1, cycle == "cycle_two")
cycle2 <- prune_taxa(taxa_sums(cycle2) > 0, cycle2)
cycle2.1 <- as(sample_data(cycle2), "data.frame")

cycle3b = subset_samples(compare1, cycle == "cycle_three_before_stress")
cycle3b <- prune_taxa(taxa_sums(cycle3b) > 0, cycle3b)
cycle3b.1 <- as(sample_data(cycle3b), "data.frame")

#cycle 1

subs <- subset_samples(cycle1, treatment2 %in% c("Control", "Chlorothalonil"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment2,data = metadata, perm=999)

subs <- subset_samples(cycle1, treatment2 %in% c("Control", "Glyphosate"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment2, data = metadata, perm=999)


subs <- subset_samples(cycle1, treatment2 %in% c("Control", "Tetracycline"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment2,data = metadata, perm=999)

pvalues <- c(0.21,0.022,0.004)
p.adjust(pvalues,method="fdr")

#cycle 2
subs <- subset_samples(cycle2, treatment2 %in% c("Control", "Chlorothalonil"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment2,data = metadata, perm=999)

subs <- subset_samples(cycle2, treatment2 %in% c("Control", "Glyphosate"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment2,data = metadata, perm=999)

subs <- subset_samples(cycle2, treatment2 %in% c("Control", "Tetracycline"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment2,data = metadata, perm=999)

pvalues <- c(0.5,0.08,0.001)
p.adjust(pvalues,method="fdr")


#cycle 3 before stress

subs <- subset_samples(cycle3b, treatment3 %in% c("Control", "Chlorothalonil"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment3,data = metadata, perm=999)

subs <- subset_samples(cycle3b, treatment3 %in% c("Control", "Glyphosate"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment3,data = metadata, perm=999)

subs <- subset_samples(cycle3b, treatment3 %in% c("Control", "Tetracycline"))
metadata <- as(sample_data(subs), "data.frame")
adonis(distance(subs, method="bray") ~ treatment3,data = metadata, perm=999)

pvalues <- c(0.158,0.002,0.001)
p.adjust(pvalues,method="fdr")
```
Community composition was significantly affected under tetracycline in all cycles. Glyphosate affected it in cycle 1 and 3 while chlorothalonil did not show to cause significant changes


#### Test general effects of treatment across cycles
```{r treatment_effects}
set.seed(42)
compare=ps1
compare <- prune_taxa(taxa_sums(compare) > 0, compare)
compare <- transform_sample_counts(compare, function(OTU) {OTU / sum(OTU)})
compare.1 <- as(sample_data(compare), "data.frame")

cycle1 = subset_samples(compare, cycle == "cycle_one")
cycle1 <- prune_taxa(taxa_sums(cycle1) > 0, cycle1)
cycle1.2 <- as(sample_data(cycle1), "data.frame")

cycle2 = subset_samples(compare, cycle == "cycle_two")
cycle2 <- prune_taxa(taxa_sums(cycle2) > 0, cycle2)
cycle2.2 <- as(sample_data(cycle2), "data.frame")

cycle3_before_stress = subset_samples(compare, cycle == "cycle_three_before_stress")
cycle3_before_stress <- prune_taxa(taxa_sums(cycle3_before_stress) > 0, cycle3_before_stress)
cycle3_before_stress.2 <- as(sample_data(cycle3_before_stress), "data.frame")

d = distance(compare, "bray")
adonis(d ~ treatment3, compare.1, perm=999)

d1 = distance(cycle1, "bray")
adonis(d1 ~ treatment3, cycle1.2, perm=999)

d2 = distance(cycle2, "bray")
adonis(d2 ~ treatment3, cycle2.2, perm=999)

d3 = distance(cycle3_before_stress, "bray")
adonis(d3 ~ treatment3, cycle3_before_stress.2, perm=999)
```

Treatment significantly explains differences in microbiome in all cycles (between 43 and 65 % of variation).

### Test for cage effects
```{r cage_effects}
set.seed(42)
cycle1_chloro = subset_samples(cycle1, treatment3 == "Chlorothalonil")
cycle1_chloro.1 <- as(sample_data(cycle1_chloro), "data.frame")
d = distance(cycle1_chloro, "bray")
adonis(d ~ cage, cycle1_chloro.1, perm=999)

cycle1_gl = subset_samples(cycle1, treatment3 == "Glyphosate")
cycle1_gl.1 <- as(sample_data(cycle1_gl), "data.frame")
d = distance(cycle1_gl, "bray")
adonis(d ~ cage, cycle1_gl.1, perm=999)

cycle1_co = subset_samples(cycle1, treatment3 == "Control")
cycle1_co.1 <- as(sample_data(cycle1_co), "data.frame")
d = distance(cycle1_co, "bray")
adonis(d ~ cage, cycle1_co.1, perm=999)

#cycle2

cycle2_control = subset_samples(cycle2, treatment3 == "Control")
cycle2_control.1 <- as(sample_data(cycle2_control), "data.frame")
d = distance(cycle2_control, "bray")
adonis(d ~ cage, cycle2_control.1, perm=999)

cycle2_chloro = subset_samples(cycle2, treatment3 == "Chlorothalonil")
cycle2_chloro.1 <- as(sample_data(cycle2_chloro), "data.frame")
d = distance(cycle2_chloro, "bray")
adonis(d ~ cage, cycle2_chloro.1, perm=999)

cycle2_gl = subset_samples(cycle2, treatment3 == "Glyphosate")
cycle2_gl.1 <- as(sample_data(cycle2_gl), "data.frame")
d = distance(cycle2_gl, "bray")
adonis(d ~ cage, cycle2_gl.1, perm=999)

cycle2_tet = subset_samples(cycle2, treatment3 == "Tetracycline")
cycle2_tet.1 <- as(sample_data(cycle2_tet), "data.frame")
d = distance(cycle2_tet, "bray")
adonis(d ~ cage, cycle2_tet.1, perm=999)

#cycle 3 before stress

cycle3_control = subset_samples(cycle3_before_stress, treatment3 == "Control")
cycle3_control.1 <- as(sample_data(cycle3_control), "data.frame")
d = distance(cycle3_control, "bray")
adonis(d ~ cage, cycle3_control.1, perm=999)

cycle3_chloro = subset_samples(cycle3_before_stress, treatment3 == "Chlorothalonil")
cycle3_chloro.1 <- as(sample_data(cycle3_chloro), "data.frame")
d = distance(cycle3_chloro, "bray")
adonis(d ~ cage, cycle3_chloro.1, perm=999)

cycle3_gl = subset_samples(cycle3_before_stress, treatment3 == "Glyphosate")
cycle3_gl.1 <- as(sample_data(cycle3_gl), "data.frame")
d = distance(cycle3_gl, "bray")
adonis(d ~ cage, cycle3_gl.1, perm=999)

cycle3_tet = subset_samples(cycle3_before_stress, treatment3 == "Tetracycline")
cycle3_tet.1 <- as(sample_data(cycle3_tet), "data.frame")
d = distance(cycle3_tet, "bray")
adonis(d ~ cage, cycle3_tet.1, perm=999)
```
Testing if the three cages within a treatment show significant variations in microbiome composition (cage effects) reveals that a cage effect is not common in our data. Only in cycle 2 control and cycle 3 glyphosate significant cage variations are seen.


### Time effect
#### Test if microbial composition of treatments differ across the three cycles

```{r}
set.seed(42)
compare = ps1
compare2 <- prune_taxa(taxa_sums(compare) > 0, compare)
compare2 <- transform_sample_counts(compare2, function(OTU) {OTU / sum(OTU)})

subs1 <- subset_samples(compare2, treatment_cycle3 == "Control_cycle_1"|treatment_cycle3=="Control_cycle_2"|treatment_cycle3=="control_cycle_3_before_stress")
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ cycle,
       data = metadata, perm=999)

subs2 <- subset_samples(compare2, treatment_cycle =="Chlorothalonil_cycle_1"|treatment_cycle== "Chlorothalonil_cycle_2"|treatment_cycle3=="Chlorothalonil_cycle_3_before_stress")
metadata <- as(sample_data(subs2), "data.frame")
adonis(distance(subs2, method="bray") ~ cycle,
       data = metadata, perm=999)

subs3 <- subset_samples(compare2, treatment_cycle3 %in% c("Glyphosate_cycle_1", "Glyphosate_cycle_2","Glyphosate_cycle_3_before_stress"))
metadata <- as(sample_data(subs3), "data.frame")
adonis(distance(subs3, method="bray") ~ cycle,
       data = metadata, perm=999)

subs4 <- subset_samples(compare2, treatment_cycle %in% c("Tetracycline_cycle_1", "Tetracycline_cycle_2","Tetracycline_cycle_3_before_stress"))
metadata <- as(sample_data(subs4), "data.frame")
adonis(distance(subs4, method="bray") ~ cycle,
       data = metadata, perm=999)
```
All treatments differ significantly across cycles

#### compare treatments before and after high stress

```{r}
set.seed(42)
compare = ps1
compare2 <- transform_sample_counts(compare2, function(OTU) {OTU / sum(OTU)})
subs1 <- subset_samples(compare2, treatment_cycle %in% c("control_cycle_3_before_stress","control_Chlorothalonil_cycle_3_after_stress"))
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment_cycle,data = metadata, perm=999)

subs1 <- subset_samples(compare2, treatment_cycle %in% c("control_cycle_3_before_stress", "control_Glyphosate_cycle_3_after_stress"))
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment_cycle,data = metadata, perm=999)

subs1 <- subset_samples(compare2, treatment_cycle %in% c("control_cycle_3_before_stress", "control_Tetracycline_cycle_3_after_stress"))
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment_cycle,data = metadata, perm=999)

subs1 <- subset_samples(compare2, treatment_cycle %in% c("Chlorothalonil_cycle_3_before_stress", "Chlorothalonil_cycle_3_after_stress"))
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment_cycle,data = metadata, perm=999)

subs1 <- subset_samples(compare2, treatment_cycle %in% c("Glyphosate_cycle_3_before_stress", "Glyphosate_cycle_3_after_stress"))
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment_cycle,data = metadata, perm=999)

subs1 <- subset_samples(compare2, treatment_cycle %in% c("Tetracycline_cycle_3_before_stress", "Tetracycline_cycle_3_after_stress"))
metadata <- as(sample_data(subs1), "data.frame")
adonis(distance(subs1, method="bray") ~ treatment_cycle,data = metadata, perm=999)
```
No significant difference for any treatment before and after high stress application -> likely the time was not enough in between the two time points and we sequenced a lot of dead bacteria



### test spread of variance of groups
```{r betadisp}
set.seed(53)
test<-ps1
test2 <- transform_sample_counts(test, function(OTU) {OTU / sum(OTU)})

cycle1 = subset_samples(test2, cycle == "cycle_one")
cycle1 <- prune_taxa(taxa_sums(cycle1) > 0, cycle1)

cycle2 = subset_samples(test2, cycle == "cycle_two")
cycle2 <- prune_taxa(taxa_sums(cycle2) > 0, cycle2)

cycle3_b = subset_samples(test2, cycle == "cycle_three_before_stress")
cycle3_b <- prune_taxa(taxa_sums(cycle3_b) > 0, cycle3_b)

#testing treatment
set.seed(53)
d_cycle1 = distance(cycle1, "bray")
df_cycle1 = as(sample_data(cycle1), "data.frame")
df_cycle1$treatment3 <- factor(df_cycle1$treatment3 , levels=c("Control", "Chlorothalonil", "Glyphosate", "Tetracycline"))
groups <- df_cycle1[["treatment3"]]
beta <- betadisper(d_cycle1, df_cycle1$treatment3)
permutest(beta,pairwise = TRUE, permutations = 999)
anova(betadisper(d_cycle1, groups))
mod.HSD <- TukeyHSD(beta,conf.level = 0.95)
mod.HSD
plot(betadisper(d_cycle1, groups),main="MultiVariate Permutation Cycle one",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
boxplot(betadisper(d_cycle1, groups),main="Cycle one",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),xlab="")

set.seed(53)
d_cycle2 = distance(cycle2, "bray")
df_cycle2 = as(sample_data(cycle2), "data.frame")
df_cycle2$treatment3 <- factor(df_cycle2$treatment3 , levels=c("Control", "Chlorothalonil", "Glyphosate", "Tetracycline"))
groups <- df_cycle2[["treatment3"]]
beta <- betadisper(d_cycle2, df_cycle2$treatment3)
permutest(beta,pairwise = TRUE, permutations = 99)
anova(betadisper(d_cycle2, groups))
mod.HSD <- TukeyHSD(beta)
mod.HSD
plot(betadisper(d_cycle2, groups),main="MultiVariate Permutation Cycle two",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
boxplot(betadisper(d_cycle2, groups),main="Cycle two",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),xlab="")

set.seed(53)
d_cycle3_b = distance(cycle3_b, "bray")
df_cycle3_b = as(sample_data(cycle3_b), "data.frame")
df_cycle3_b$treatment3 <- factor(df_cycle3_b$treatment3 , levels=c("Control", "Chlorothalonil", "Glyphosate", "Tetracycline"))
groups <- df_cycle3_b[["treatment3"]]
beta <- betadisper(d_cycle3_b, df_cycle3_b$treatment3)
permutest(beta,pairwise = TRUE, permutations = 99)
anova(betadisper(d_cycle3_b, groups))
mod.HSD <- TukeyHSD(beta)
mod.HSD
plot(betadisper(d_cycle3_b, groups),main="MultiVariate Permutation Cycle three",lwd=2, label = TRUE,label.cex = 0.7,col=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
boxplot(betadisper(d_cycle3_b, groups),main="Cycle three",border=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),xlab="")
```

# Taxonomy plots and stats

```{r set_up_for_taxa_plots}
tax <-ps1
ps_bdiv2 <- transform_sample_counts(tax, function(OTU) {OTU / sum(OTU)})
tax_all2<- ps_bdiv2
ps_pd2 <- tax_glom(tax_all2, taxrank = 'Genus')

low1pc_reads <- max(taxa_sums(ps_pd2)) * 1 / 100
low1pc_indices <- which(taxa_sums(ps_pd2) < low1pc_reads)
length(low1pc_indices)
taxa_names(ps_pd2)[low1pc_indices[1]] <- "other"
taxa3 <- merge_taxa(ps_pd2, low1pc_indices, "other")
tax_table(taxa3)["other", ] <- c("other")

#build a custom color palette for phyloseq object
getPalette = colorRampPalette(brewer.pal(8, "Set3"))
speciesList = unique(tax_table(taxa3)[,"Genus"])
speciesPalette = getPalette(length(speciesList))
names(speciesPalette) = speciesList
```

```{r taxa_plot}
ps2 <-taxa3
ps3 = subset_samples(ps2, cycle=="cycle_one" | cycle=="cycle_two" | cycle == "cycle_three_before_stress")
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)

ps_df <- psmelt(ps3)
ps_df_sum <- ps_df %>%
  group_by(treatment3,Genus,cycle) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment3,cycle) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps_df_sum, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)

p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
#p<-p+ theme(legend.position="bottom")
p<-p+ theme(legend.text=element_text(size=13, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),
                                                     legend.key.size = unit(0.6, "cm"))+theme(legend.title = element_blank())
#p<-p + facet_grid( ~cycle, scales="free_x",space="free")
p<-p + facet_wrap( ~cycle, scales="free_x",nrow=2)
p<-p + theme(strip.text = element_text(size=14,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=14, face="bold"))+theme(axis.text.y = element_text(size=13, face="bold"))+theme(axis.text.x = element_text(size=10, face="bold", angle = 90))+theme(axis.title.x = element_blank())+theme(axis.text.x = element_text(vjust = 0.5,hjust=1))+theme(strip.background =element_rect(fill="white", color="white"))+theme(axis.text.x =element_blank())#+theme(axis.text.x = element_text(vjust = 0.5,hjust=1))
taxa2
ggsave("R_microbiome_figures/taxa_man.png", height = 4.5, width = 6.5)
```

```{r taxa_plot2}
ps2 <-taxa3
ps3 = subset_samples(ps2, cycle !="positive_control" & cycle !="hive_bee" & cycle !="cycle_two"& cycle !="cycle_one"& cycle !="start_microbiome")
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)

ps_df <- psmelt(ps3)
ps_df_sum <- ps_df %>%
  group_by(treatment3,Genus,cycle) %>%
  summarise(Abundance = mean(Abundance)) %>%
  group_by(treatment3,cycle) %>%
  mutate(relative_abundance = Abundance / sum(Abundance)) %>%
  ungroup() %>%
  mutate(Genus = fct_reorder(Genus, Abundance, .fun = sum, .desc = TRUE))
levels(ps_df_sum$Genus)

neworder <- c("cycle_three_before_stress","cycle_three_after_stress")
change <- c("Cycle three before stress","Cycle three after stress")

ps_df_sum2 <- arrange(transform(ps_df_sum, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","control_Chlorothalonil", "Chlorothalonil","control_Glyphosate","Glyphosate","control_Tetracycline", "Tetracycline")
change2 <- c("Control","control \n Chlorothalonil", "Chlorothalonil","control \n Glyphosate","Glyphosate","control \n Tetracycline", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2,labels=change2)),treatment3)

p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = relative_abundance, fill = Genus)) +
  geom_bar(stat = "identity")+ scale_fill_manual(values= speciesPalette)+ scale_y_continuous(expand = c(0,0))+ylab("relative abundance")
p<-p+ theme(legend.position="bottom")
p<-p+ theme(legend.text=element_text(size=10, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),
                                                     legend.key.size = unit(0.6, "cm"))+theme(legend.title = element_blank())+ facet_grid( ~cycle, scales="free_x",space="free")
taxa3<-p +theme(strip.text = element_text(size=12,face="bold",color="grey40"))
taxa3<-taxa3+theme(axis.title.y = element_text(size=11, face="bold"))+theme(axis.text.y = element_text(size=10, face="bold"))+theme(axis.text.x = element_text(size=10, face="bold", angle = 90))+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))+theme(axis.text.x = element_text(vjust = 0.5,hjust=1))
taxa3
ggsave("R_microbiome_figures/Supp_taxonomy.png", height = 4.5, width = 5.9)
```

## Core taxa abundance

```{r total abundance}
ps2 <-ps1
ps2 <- rarefy_even_depth(ps2,sample.size=12351, replace=FALSE, rngseed = 1) 
ps2 = subset_samples(ps2, cycle == "cycle_one"| cycle == "cycle_two"| cycle == "cycle_three_before_stress")
ps3 <- prune_taxa(taxa_sums(ps2) > 0, ps2)
ps3 <- tax_glom(ps3, taxrank = 'Genus')
psOrd3 = subset_taxa(ps3, Genus=="Lactobacillus" | Genus=="Bartonella" | Genus=="Gilliamella" | Genus=="Snodgrassella" | Genus=="Frischella" | Genus=="Bifidobacterium" | Genus=="Commensalibacter")

#Melt and plot
melt<-psmelt(psOrd3)
levels2=c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
melt2 <- arrange(transform(melt, cycle=factor(cycle,levels=levels2,labels=change)),cycle)
neworder2 <- c("Bartonella","Lactobacillus","Gilliamella","Snodgrassella","Bifidobacterium","Commensalibacter","Frischella")
melt3 <- arrange(transform(melt2, Genus=factor(Genus,levels=neworder2)),Genus)
neworder3 <- c("Control","Chlorothalonil","Glyphosate","Tetracycline")
melt4 <- arrange(transform(melt3, treatment3=factor(treatment3,levels=neworder3)),treatment3)

a_mean <- melt4 %>% 
  group_by(treatment3,Genus,cycle) %>% 
  summarize(mean_val = mean(Abundance))
a_mean2<-subset(a_mean, treatment3 =="Control")
print(a_mean2)

sigFunc = function(x){
  if(x < 0.001){"***"} 
  else if(x < 0.01){"**"}
  else if(x < 0.05){"*"}
  else{NA}}

p<-ggplot(data = melt4, aes(x = treatment3, y = Abundance)) +
  geom_boxplot(aes(fill=Genus),alpha=0.5,lwd=0.7, position = position_dodge(width = 0.3), width=0.45,outlier.shape = NA) + geom_jitter(aes(color = Genus), height = 0, width = .1)+geom_hline(data= a_mean2, aes(yintercept=mean_val), width=5,size=1,linetype="dashed")+
  labs(x = "", y = "Abundance\n")+
  facet_grid(Genus~cycle, scales = "free")+theme_bw()+
  geom_signif(comparisons=list(c("Control", "Chlorothalonil"), c("Control", "Glyphosate"),c("Control", "Tetracycline")),tip_length = 0.03,vjust=1.3, color="grey20", size=1.1,map_signif_level=sigFunc, textsize=9, test = wilcox.test,step_increase = 0.2)
p<-p+ theme(legend.position="right")+ylab("Total abundance")
p<-p+ theme(legend.text=element_text(size=29, face = "italic"))+theme(legend.key = element_rect(color = NA, fill = NA),legend.key.size = unit(0.9, "cm"))+theme(legend.title = element_blank())
abu<-p +theme(strip.text.x = element_text(size=25,face="bold",color = "grey 35"))+theme(strip.background =element_rect(fill="white", color="white"))
cycle<-abu+theme(axis.title.y = element_text(size=25, face="bold"))+theme(axis.text.y = element_text(size=20, face="bold"))+theme(axis.text.x = element_text(size=25, angle=90,face="bold"))+theme(axis.title.x = element_blank())+ theme(strip.text.y = element_blank())
ggsave("R_microbiome_figures/core_total_abundance.png", height = 18, width = 16)
``` 

### stats on abundances

### compare if taxa abundances are significantly different between controls and treatments in cycle 1, 2 and cycle 3, use Wilcoxon tests with following fdr correction. In addition test if control samples differ in the abundances of the core taxa between cycle 1 and 3 (time or lab-adaptation effect)

```{r, lactobacillus_differ}
set.seed(42)
ps2 <-ps1
ps2 <- rarefy_even_depth(ps2,sample.size=12351, replace=FALSE, rngseed = 1) 
ps2 = subset_samples(ps2, cycle == "cycle_one"| cycle == "cycle_two"| cycle == "cycle_three_before_stress")
ps3 <- prune_taxa(taxa_sums(ps2) > 0, ps2)
abundance <- tax_glom(ps3, taxrank = 'Genus')
Lac = subset_taxa(abundance, Genus=="Lactobacillus")

#cycle 1
Gly <- subset_samples(Lac, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Lac, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Lac, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.9774,0.4428,0.0044)
p.adjust(pvalues,method="fdr")

#cycle 2
Gly <- subset_samples(Lac, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Lac, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Lac, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.45,0.67,0.000022)
p.adjust(pvalues,method="fdr")

#cycle 3
Gly <- subset_samples(Lac, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Lac, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Lac, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.136,0.75,0.0007)
p.adjust(pvalues,method="fdr")


# test control cycle 1 in comparison to cycle 3

Lac1 <- subset_samples(Lac,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Lac1 <- prune_taxa(taxa_sums(Lac1) > 0, Lac1)
melt<-psmelt(Lac1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```

```{r, Bartonella_differ}
Bart = subset_taxa(abundance, Genus=="Bartonella")

Gly <- subset_samples(Bart, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Bart, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Bart, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.76,0.55,0.0044)
p.adjust(pvalues,method="fdr")

Gly <- subset_samples(Bart, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Bart, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Bart, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.65, 0.48, 5.253e-05)
p.adjust(pvalues,method="fdr")


#cycle 3
Gly <- subset_samples(Bart, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Bart, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Bart, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.3,0.94,0.0004)
p.adjust(pvalues,method="fdr")


# test control cycle 1 in comparison to cycle 3
Bart1 <- subset_samples(Bart,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Bart1 <- prune_taxa(taxa_sums(Bart1) > 0, Bart1)
melt<-psmelt(Bart1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```
#### Gilliamella
```{r,Gilliamella_abund}
Gil = subset_taxa(abundance, Genus=="Gilliamella")

Gly <- subset_samples(Gil, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Gil, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Gil, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.07,0.06,0.0044)
p.adjust(pvalues,method="fdr")

Gly <- subset_samples(Gil, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Gil, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Gil, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.03,0.55,0.003)
p.adjust(pvalues,method="fdr")

#cycle 3
Gly <- subset_samples(Gil, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Gil, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Gil, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.3707,0.6406,2.124e-08)
p.adjust(pvalues,method="fdr")

#cycle 3 in comp cycle 1 control
Gil1 <- subset_samples(Gil,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Gil1 <- prune_taxa(taxa_sums(Gil1) > 0, Gil1)
melt<-psmelt(Gil1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```
#### Snodgrassella
```{r,Snodgrassella_abund}
set.seed(42)
ps2 <-ps1
ps2 <- rarefy_even_depth(ps2,sample.size=12351, replace=FALSE, rngseed = 1) 
ps2 = subset_samples(ps2, cycle == "cycle_one"| cycle == "cycle_two"| cycle == "cycle_three_before_stress")
ps3 <- prune_taxa(taxa_sums(ps2) > 0, ps2)
abundance <- tax_glom(ps3, taxrank = 'Genus')
Snod = subset_taxa(abundance, Genus=="Snodgrassella")

Gly <- subset_samples(Snod, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Snod, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Snod, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.0002,0.8,0.0113)
p.adjust(pvalues,method="fdr")

Gly <- subset_samples(Snod, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Snod, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Snod, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(1.479e-06,0.16,3.588e-05)
p.adjust(pvalues,method="fdr")

#cycle 3
Gly <- subset_samples(Snod, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Snod, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Snod, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.42,0.78,9.578e-06)
p.adjust(pvalues,method="fdr")

#cycle 3 in comp cycle 1 control
Snod1 <- subset_samples(Snod,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Snod1 <- prune_taxa(taxa_sums(Snod1) > 0, Snod1)
melt<-psmelt(Snod1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```

#### Frischella
```{r,Frischella_abund}
Fri = subset_taxa(abundance, Genus=="Frischella")

Gly <- subset_samples(Fri, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Fri, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Fri, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.06872,0.033,0.004396)
p.adjust(pvalues,method="fdr")

Gly <- subset_samples(Fri, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Fri, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Fri, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.7818,0.713,0.00001831)
p.adjust(pvalues,method="fdr")


#cycle 3
Gly <- subset_samples(Fri, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Fri, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Fri, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.02,0.64,8.806e-06)
p.adjust(pvalues,method="fdr")

#cycle 3 in comp cycle 1 control
Fri1 <- subset_samples(Fri,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Fri1 <- prune_taxa(taxa_sums(Fri1) > 0, Fri1)
melt<-psmelt(Fri1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```

#### Commensalibacter
```{r,Commensalibacter_abund}
Com = subset_taxa(abundance, Genus=="Commensalibacter")

Gly <- subset_samples(Com, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Com, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Com, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.069,0.795,0.0111)
p.adjust(pvalues,method="fdr")

Gly <- subset_samples(Com, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Com, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Com, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.288,0.149,0.0001015)
p.adjust(pvalues,method="fdr")

#cycle 3
Gly <- subset_samples(Com, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Com, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Com, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.41,1,0.000012)
p.adjust(pvalues,method="fdr")

#cycle 3 in comp cycle 1 control
Com1 <- subset_samples(Com,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Com1 <- prune_taxa(taxa_sums(Com1) > 0, Com1)
melt<-psmelt(Com1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```
#### Bifidobacterium
```{r,Bifidobacterium_abund}
Bif = subset_taxa(abundance, Genus=="Bifidobacterium")

Gly <- subset_samples(Bif, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Glyphosate_cycle_1")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Bif, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Chlorothalonil_cycle_1")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Bif, treatment_cycle =="Control_cycle_1" | treatment_cycle =="Tetracycline_cycle_1")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.755,0.887,0.9451)
p.adjust(pvalues,method="fdr")

Gly <- subset_samples(Bif, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Glyphosate_cycle_2")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment2, data=melt)

Chlo <- subset_samples(Bif, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Chlorothalonil_cycle_2")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment2, data=melt)

Tet <- subset_samples(Bif, treatment_cycle =="Control_cycle_2" | treatment_cycle =="Tetracycline_cycle_2")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment2, data=melt)

pvalues<-c(0.5254,0.603,0.065)
p.adjust(pvalues,method="fdr")

#cycle 3
Gly <- subset_samples(Bif, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Glyphosate_cycle_3_before_stress")
Gly <- prune_taxa(taxa_sums(Gly) > 0, Gly)
melt<-psmelt(Gly)
wilcox.test(Abundance~treatment3, data=melt)

Chlo <- subset_samples(Bif, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Chlorothalonil_cycle_3_before_stress")
Chlo <- prune_taxa(taxa_sums(Chlo) > 0, Chlo)
melt<-psmelt(Chlo)
wilcox.test(Abundance~treatment3, data=melt)

Tet <- subset_samples(Bif, treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Tetracycline_cycle_3_before_stress")
Tet <- prune_taxa(taxa_sums(Tet) > 0, Tet)
melt<-psmelt(Tet)
wilcox.test(Abundance~treatment3, data=melt)

pvalues<-c(0.093,0.802,0.298)
p.adjust(pvalues,method="fdr")

#cycle 3 in comp cycle 1 control
Bif1 <- subset_samples(Bif,treatment_cycle=="control_cycle_3_before_stress"|treatment_cycle=="Control_cycle_1")
Bif1 <- prune_taxa(taxa_sums(Bif1) > 0, Bif1)
melt<-psmelt(Bif1)
wilcox.test(Abundance~treatment_cycle, data=melt)
```

### core taxa abundances: Gardner-Altman estimation plots

```{r, dabest_abundance_init}
#devtools::install_github("mikheyev/dabestr")
library(dabestr)
microbeColors <- function (n) {
  mcols <- c("#55596a", "#6ebe9f","#f3a935", "#D45E79")
  return(mcols[4 %/% n])
}
```
#### Bartonella
```{r, dabest_abundance_Bartonella}
ps2 <-ps1
set.seed(42)
ps2.rare = rarefy_even_depth(ps2, rngseed=1, sample.size=12351, replace=F)

sample_data(ps2.rare)$treatment3<-factor(sample_data(ps2.rare)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2.rare)$treatment3)
cycle1 = subset_samples(ps2.rare, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Bartonella") 

#Melt and plot
melt<-psmelt(psord)

unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)

# Display the results in a user-friendly format.
unpaired_mean_diff
Bartonella_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=17,axes.title.fontsize=19,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2.rare, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Bartonella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Bartonella_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))


#cycle3
cycle3 = subset_samples(ps2.rare, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Bartonella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Bartonella_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

#### Lactobacillus
```{r, dabest_abundance_Lactobacillus}
ps2 <-ps1
set.seed(42)
ps2.rare = rarefy_even_depth(ps2, rngseed=1, sample.size=12351, replace=F)
sample_data(ps2.rare)$treatment3<-factor(sample_data(ps2.rare)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2.rare)$treatment3)
cycle1 = subset_samples(ps2.rare, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
psord = subset_taxa(ps3, Genus=="Lactobacillus") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Lactobacillus_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=17,axes.title.fontsize=19,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2.rare, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Lactobacillus") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Lactobacillus_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle3
cycle3 = subset_samples(ps2.rare, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Lactobacillus") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Lactobacillus_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

#### Snodgrassella
```{r, dabest_abundance_Snodgrassella}
ps2 <-ps1
set.seed(42)
ps2.rare = rarefy_even_depth(ps2, rngseed=1, sample.size=12351, replace=F)
sample_data(ps2.rare)$treatment3<-factor(sample_data(ps2.rare)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2.rare)$treatment3)
cycle1 = subset_samples(ps2.rare, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Snodgrassella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Snodgrassella_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=17,axes.title.fontsize=19,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2.rare, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Snodgrassella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Snodgrassella_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle3
cycle3 = subset_samples(ps2.rare, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Snodgrassella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Snodgrassella_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

#### Gilliamella
```{r, dabest_abundance_Gilliamella}
ps2 <-ps1
sample_data(ps2)$treatment3<-factor(sample_data(ps2)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2)$treatment3)
cycle1 = subset_samples(ps2, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Gilliamella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Gilliamella_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=15.5,axes.title.fontsize=17.5,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Gilliamella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Gilliamella_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=15.5,axes.title.fontsize=17.5,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle3
cycle3 = subset_samples(ps2, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Gilliamella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Gilliamella_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=15.5,axes.title.fontsize=17.5, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

#### Bifidobacterium
```{r, dabest_abundance_Bifidobacterium}
ps2 <-ps1
set.seed(42)
ps2.rare = rarefy_even_depth(ps2, rngseed=1, sample.size=12351, replace=F)
sample_data(ps2.rare)$treatment3<-factor(sample_data(ps2.rare)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2.rare)$treatment3)
cycle1 = subset_samples(ps2.rare, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Bifidobacterium") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Bifidobacterium_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=17,axes.title.fontsize=19,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2.rare, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Bifidobacterium") 

#Melt and plot
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Bifidobacterium_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle3
cycle3 = subset_samples(ps2.rare, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Bifidobacterium") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Bifidobacterium_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

#### Frischella
```{r, dabest_abundance_Frischella}
ps2 <-ps1
set.seed(42)
ps2.rare = rarefy_even_depth(ps2, rngseed=1, sample.size=12351, replace=F)

sample_data(ps2.rare)$treatment3<-factor(sample_data(ps2.rare)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2.rare)$treatment3)
cycle1 = subset_samples(ps2.rare, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Frischella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Frischella_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=17,axes.title.fontsize=19,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2.rare, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Frischella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Frischella_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle3
cycle3 = subset_samples(ps2.rare, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Frischella") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Frischella_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

#### Commensalibacter
```{r, dabest_abundance_Commensalibacter}
ps2 <-ps1
set.seed(42)
ps2.rare = rarefy_even_depth(ps2, rngseed=1, sample.size=12351, replace=F)

sample_data(ps2.rare)$treatment3<-factor(sample_data(ps2.rare)$treatment3,levels=c("Control","Chlorothalonil","Glyphosate","Tetracycline"), labels=c("Control","Chloro","Glypho","Tetra"))
levels(sample_data(ps2.rare)$treatment3)
cycle1 = subset_samples(ps2.rare, cycle == "cycle_one")
ps3 <- tax_glom(cycle1, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Commensalibacter") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Commensalibacter_cycle1<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",tick.fontsize=17,axes.title.fontsize=19,effsize.ylabel = "Mean diff.", rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle2
cycle2 = subset_samples(ps2.rare, cycle == "cycle_two")
ps3 <- tax_glom(cycle2, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Commensalibacter") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Commensalibacter_cycle2<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19,rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))

#cycle3
cycle3 = subset_samples(ps2.rare, cycle == "cycle_three_before_stress")
ps3 <- tax_glom(cycle3, taxrank = 'Genus')
ps3 <- prune_taxa(taxa_sums(ps3) > 0, ps3)
psord = subset_taxa(ps3, Genus=="Commensalibacter") 
melt<-psmelt(psord)
unpaired_mean_diff <- dabest(melt, treatment3,Abundance,
                             idx = c("Control", "Chloro", "Glypho","Tetra"),
                             paired = FALSE)
unpaired_mean_diff
Commensalibacter_cycle3<-plot(unpaired_mean_diff, rawplot.ylabel = "Abundance",effsize.ylabel = "Mean diff.",tick.fontsize=17,axes.title.fontsize=19, rawplot.markersize = 2, colorValues = c("#55596a", "#6ebe9f","#f3a935", "#D45E79"))
```

### have a look on most abundant ASVs in the core taxa

```{r Gilliamella_ASV}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Gilliamella")

ps2.3 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2.4 <- subset_taxa(ps2.3, ASV !="ASV0068" & ASV !="ASV0070")

ps2m <- psmelt(ps2.4)

neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)

p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~ASV, scales="free_y",nrow=2)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Gilliamella apicola")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/ASV_Gilliamella.png", height = 4.5, width = 10)
```

```{r Bifidobacterium_ASV}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Bifidobacterium")

ps2.3 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2m <- psmelt(ps2.3)

neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)

p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~ASV, scales="free_y",nrow=2)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Bifidobacterium sp.")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/ASV_Bifidobacterium.png", height = 4.5, width = 7)
```

### species Lactobacillus
```{r Lacto_species}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Lactobacillus")
ps2 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2 <- tax_glom(ps2, taxrank = 'Species')
ps2m <- psmelt(ps2)

neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)

neworder3 <- c("Lactobacillus apis","Lactobacillus kullabergensis","Lactobacillus helsingborgensis","Lactobacillus kunkeei","Lactobacillus sp. G3_4_3CO2","Lactobacillus melliventris","not_available")
change3 <- c("L. apis","L. kullabergensis","L. helsingborgensis","L. kunkeei","L. kimbladii","L. melliventis","not_available")
ps_df_sum4 <- arrange(transform(ps_df_sum3, Species=factor(Species,levels=neworder3,labels=change3)),Species)

p<-ggplot(data = ps_df_sum4, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~Species, scales="free_y",nrow=3)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Lactobacillus spp.")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/Lactobacillus spp.png", height = 7, width = 11)
```

### ASV Lactobacillus
```{r ASV_L.apis}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Lactobacillus")
ps2 <- subset_taxa(ps2, Species =="Lactobacillus apis")
ps2.3 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2.3 <- prune_taxa(taxa_sums(ps2.3) > 0, ps2.3)
ps2m <- psmelt(ps2.3)
neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)
neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)
p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~ASV, scales="free_y",nrow=1)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Lactobacillus apis")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/ASV_Lactobacillus apis.png", height = 3, width = 8)
```
# Lactobacillus kullabergensis
```{r ASV_L.kullabergensis}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Lactobacillus")
ps2 <- subset_taxa(ps2, Species =="Lactobacillus kullabergensis")
ps2.3 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2.3 <- prune_taxa(taxa_sums(ps2.3) > 0, ps2.3)
ps2m <- psmelt(ps2.3)
neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)

p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~ASV, scales="free_y",nrow=1)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Lactobacillus kullabergensis")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/ASV_Lactobacillus kullabergensis.png", height = 3, width = 8)
```

# Lactobacillus NA
```{r ASV_L.NA}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Lactobacillus")
ps2 <- subset_taxa(ps2, Species =="not_available")
ps2 <- subset_taxa(ps2, ASV !="ASV0069")
ps2.3 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2m <- psmelt(ps2.3)
neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)
neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)
neworder3 <- c("ASV0004","ASV0005","ASV0009", "ASV0027","ASV0031","ASV0045","ASV0052","ASV0067","ASV0030","ASV0033","ASV0035","ASV0036","ASV0043","ASV0044","ASV0063","ASV0019","ASV0037")
ps_df_sum4 <- arrange(transform(ps_df_sum3, ASV=factor(ASV,levels=neworder3)),ASV)

p<-ggplot(data = ps_df_sum4, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~ASV, scales="free_y",nrow=4)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Lactobacillus sp. NA")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/ASV_Lactobacillus sp. NA.png", height = 8, width = 13)
```

### Snodgrassella
```{r Snodgrassella_ASV}
tax<-ps1
tax = rarefy_even_depth(tax, rngseed=1, sample.size=12351, replace=F)
ps2 <- subset_samples(tax, cycle =="cycle_one"|cycle =="cycle_two"|cycle =="cycle_three_before_stress")
ps2 <- subset_samples(ps2, treatment3 =="Control"|treatment3 =="Chlorothalonil"|treatment3 =="Glyphosate"|treatment3 =="Tetracycline")
ps2 <- subset_taxa(ps2, Genus =="Snodgrassella")
ps2.3 <- prune_taxa(taxa_sums(ps2) > 1000, ps2)
ps2m <- psmelt(ps2.3)
neworder <- c("cycle_one","cycle_two","cycle_three_before_stress")
change <- c("Cycle one","Cycle two","Cycle three")
ps_df_sum2 <- arrange(transform(ps2m, cycle=factor(cycle,levels=neworder,labels=change)),cycle)

neworder2 <- c("Control","Chlorothalonil","Glyphosate", "Tetracycline")
ps_df_sum3 <- arrange(transform(ps_df_sum2, treatment3=factor(treatment3,levels=neworder2)),treatment3)

p<-ggplot(data = ps_df_sum3, aes(x = treatment3, y = Abundance, colour = treatment3)) +
  geom_point(position = position_jitter())+geom_boxplot(show.legend=FALSE, alpha = 0,colour="grey53",size=0.7)+scale_color_manual(values=c("#55596a", "#6ebe9f","#f3a935", "#D45E79"),guide = guide_legend(override.aes = list(shape = c(rep(15, 4)),size=7,alpha=0.5)))
p<-p + facet_wrap(~ASV, scales="free_y",nrow=2)
p<-p + theme(strip.text = element_text(size=13,face="bold",color="grey40"))
taxa2<-p+theme(axis.title.y = element_text(size=13, face="bold"))+theme(axis.text.y = element_text(size=12, face="bold"))+theme(axis.text.x = element_blank())+theme(axis.title.x = element_blank())+theme(strip.background =element_rect(fill="white", color="white"))
taxa2<- taxa2 + theme(legend.title = element_blank(),legend.background = element_blank(),legend.key = element_blank(),legend.text = element_text(size=14, face="italic"))
taxa2+ggtitle("Snodgrassella")+theme(plot.title = element_text(hjust = 0.5, size=14, face="bold.italic",colour = "grey38"))
ggsave("R_microbiome_figures/ASV_Snodgrassella.png", height = 6, width = 10.5)
```